File: app/src/main/java/com/jlbc/mathabs/ProblemActivity.java
package com.jlbc.mathabs;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.KeyEvent;
import android.view.inputmethod.EditorInfo;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class ProblemActivity extends AppCompatActivity
{
    private final Problem[] problems = new Problem[10];
    private final long[] times = new long[10];
    private Problem problem;
    private TextView viewProblem;
    private EditText editAnswer;
    private long start;
    private int i = 0;
    private int correct = 0;
    private InputMethodManager imm;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        for (int i = 0; i < 10; i++)
        {
            problems[i] = new Problem();
            Log.i("info", problems[i].toString());
        }
        setContentView(R.layout.activity_problem);

        viewProblem = findViewById(R.id.textProblem);
        editAnswer = findViewById(R.id.editAnswer);

        imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
        editAnswer.setOnEditorActionListener(new TextView.OnEditorActionListener() {
            @Override
            public boolean onEditorAction(TextView textView, int actionID, KeyEvent keyEvent) {
                switch(actionID)
                {
                    case EditorInfo.IME_ACTION_DONE:
                    case EditorInfo.IME_ACTION_NEXT:
                    case EditorInfo.IME_ACTION_PREVIOUS:
                        times[i] = System.currentTimeMillis() - start;
                        problem.setAnswer(Integer.parseInt(editAnswer.getText().toString()));
                        correct += problem.getResult() ? 1 : 0;
                        if (i < 9) {
                            i++;
                            displayProblem();
                            imm.showSoftInput(editAnswer, InputMethodManager.SHOW_FORCED);
                        }
                        else {
                            Toast.makeText(ProblemActivity.this, "Correct: " + correct, Toast.LENGTH_SHORT).show();
                        }
                        return true;
                }
                return true;
            }
        });
    }

    @Override
    protected void onResume()
    {
        super.onResume();
        displayProblem();
        imm.showSoftInput(editAnswer, InputMethodManager.SHOW_FORCED);
    }

    private void displayProblem()
    {
        problem = problems[i];
        viewProblem.setText(String.format("%d %s %d", problem.getA(), problem.getOperator(), problem.getB()));
        editAnswer.setText("");
        editAnswer.requestFocus();
        start = System.currentTimeMillis();
    }
}

File: app/src/main/java/com/jlbc/mathabsk/ProblemActivity.kt
package com.jlbc.mathabsk

import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.inputmethod.EditorInfo
import android.view.inputmethod.InputMethodManager
import android.widget.EditText
import android.widget.TextView
import android.widget.TextView.OnEditorActionListener
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class ProblemActivity : AppCompatActivity() {

    private val problems = arrayOfNulls<Problem>(10)
    private val times = LongArray(10)
    private var problem: Problem? = null
    private var viewProblem: TextView? = null
    private var editAnswer: EditText? = null
    private var imm: InputMethodManager? = null
    private var start: Long = 0
    private var i = 0
    private var correct = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        for (i in 0..9) {
            problems[i] = Problem()
            Log.i("info", problems[i].toString())
        }
        setContentView(R.layout.activity_problem)

        viewProblem = findViewById<View>(R.id.textProblem) as TextView
        editAnswer = findViewById<View>(R.id.editAnswer) as EditText
        imm = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
        editAnswer!!.setOnEditorActionListener(OnEditorActionListener { _, actionID, _ ->
            when (actionID) {
                EditorInfo.IME_ACTION_DONE, EditorInfo.IME_ACTION_NEXT, EditorInfo.IME_ACTION_PREVIOUS -> {
                    times[i] = System.currentTimeMillis() - start
                    problem!!.answer = editAnswer!!.text.toString().toInt()
                    correct += if (problem!!.result) 1 else 0
                    if (i < 9) {
                        i++
                        displayProblem()
                        imm!!.showSoftInput(editAnswer, InputMethodManager.SHOW_FORCED)
                    } else {
                        Toast.makeText(
                            this@ProblemActivity,
                            "Correct: $correct",
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                    return@OnEditorActionListener true
                }
            }
            true
        })
    }

    override fun onResume() {
        super.onResume()
        displayProblem()
        imm!!.showSoftInput(editAnswer, InputMethodManager.SHOW_FORCED)
    }

    private fun displayProblem() {
        problem = problems[i]
        viewProblem!!.text = "${problem!!.a} ${problem!!.operator} ${problem!!.b}"
        editAnswer!!.setText("")
        editAnswer!!.requestFocus()
        start = System.currentTimeMillis()
    }

}